### PR TITLE
Fixing issues created by #40

### PR DESCRIPTION
**Pull request by cutandpastey**
This PR should fix the following issues:

https://github.com/guardian/scribe-plugin-noting/issues/41
https://github.com/guardian/scribe-plugin-noting/issues/42
https://github.com/guardian/scribe-plugin-noting/issues/44
https://github.com/guardian/scribe-plugin-noting/issues/45

### QA
Try splitting a note into two and then un-note the first note. This should un-note ONLY the first note and leave the second untouched.

Also split one note into to and check they both have distinct data-note-id's

### CODE REVIEW

File: src/actions/noting/unwrap-note.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 var _ = require('lodash');
 var isVFocus = require('../../utils/vfocus/is-vfocus');
-var isNote = require('../../utils/noting/is-note');
+var isNote = require('../../utils/noting/is-note-segment');
```
Comment (robinedman) [inline src/actions/noting/unwrap-note.js]: ` var isNoteSegment` instead?

File: src/api/note-vdom.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ var hasAttribute = require('../utils/vdom/has-attribute');
 var isTag = require('../utils/vdom/is-tag');
 var isEmpty = require('../utils/vdom/is-empty');
 
-var focusOnNote = require('../utils/noting/is-note');
+var focusOnNote = require('../utils/noting/is-note-segment');
```
Comment (robinedman) [inline src/api/note-vdom.js]: required file and variable have different names

Comment (robinedman) [inline src/api/note-vdom.js]: Actually for quite a few of these. But I guess you haven't got around to refactoring these yet?

File: src/utils/noting/find-all-notes.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 var _ = require('lodash');
 var isVFocus = require('../vfocus/is-vfocus');
-var isNote = require('./is-note');
+var isNote = require('./is-note-segment');
```
Comment (robinedman) [inline src/utils/noting/find-all-notes.js]: `isNoteSegment`?

File: src/utils/noting/find-entire-note.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 var isVFocus = require('../vfocus/is-vfocus');
 var isWithinNote = require('./is-within-note');
-var isNote = require('./is-note');
-var findFirstNote = require('./find-first-note');
+var isNote = require('./is-note-segment');
```
Comment (robinedman) [inline src/utils/noting/find-entire-note.js]: `isNoteSegment`?

File: src/utils/noting/find-entire-note.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
+var findFirstNote = require('./find-first-note-segment');
```
Comment (robinedman) [inline src/utils/noting/find-entire-note.js]: `findFirstNoteSegment`?

Comment (robinedman) [inline src/utils/noting/find-entire-note.js]: There are some more similar ones elsewhere where the variable is lacking `Segment` at the end of the name.

File: src/api/note-vdom.js (hunk 2 above)
Comment (cutandpastey) [inline src/api/note-vdom.js]: They have now been changed. Doesn't make sense to change this one however as we will be removing this file soon :)

File: src/utils/noting/find-entire-note.js [hunk 6]
```diff
@@ -23,12 +23,12 @@ module.exports = function findEntireNote(focus) {
     throw new Error('Only a valid VFocus element can be passed to findEntirenote');
   }
 
-  var parentNote = findFirstNote(focus);
+  var parentNote = findFirstNoteSegment(focus);
```
Comment (robinedman) [inline src/utils/noting/find-entire-note.js]: This seems a bit odd.

File: test/unit/utils/noting/find-first-note-segment.spec.js [hunk 7]
```diff
@@ -5,7 +5,7 @@ var expect = chai.expect;
 var h = require('virtual-hyperscript');
 
 var VFocus = require(path.resolve(process.cwd(), 'src/vfocus'));
-var findFirstNote = require(path.resolve(process.cwd(), 'src/utils/noting/find-first-note'));
+var findFirstNote = require(path.resolve(process.cwd(), 'src/utils/noting/find-first-note-segment'));
```
Comment (robinedman) [inline test/unit/utils/noting/find-first-note-segment.spec.js]: Lacking `Segment`

File: test/unit/utils/noting/find-last-note-segment.spec.js [hunk 8]
```diff
@@ -5,7 +5,7 @@ var expect = chai.expect;
 var h = require('virtual-hyperscript');
 
 var VFocus = require(path.resolve(process.cwd(), 'src/vfocus'));
-var findLastNote = require(path.resolve(process.cwd(), 'src/utils/noting/find-last-note'));
+var findLastNote = require(path.resolve(process.cwd(), 'src/utils/noting/find-last-note-segment'));
```
Comment (robinedman) [inline test/unit/utils/noting/find-last-note-segment.spec.js]: Lacking `Segment`

File: test/unit/utils/noting/find-parent-note-segment.spec.js [hunk 9]
```diff
@@ -4,7 +4,7 @@ var expect = chai.expect;
 
 var h = require('virtual-hyperscript');
 
-var findParentNote = require(path.resolve(process.cwd(), 'src/utils/noting/find-parent-note'));
+var findParentNote = require(path.resolve(process.cwd(), 'src/utils/noting/find-parent-note-segment'));
```
Comment (robinedman) [inline test/unit/utils/noting/find-parent-note-segment.spec.js]: Lacking `Segment`

File: test/unit/utils/noting/is-note-segment.spec.js [hunk 10]
```diff
@@ -5,7 +5,7 @@ var h = require('virtual-hyperscript');
 
 var VFocus = require(path.resolve(process.cwd(), 'src/vfocus'));
 
-var isNote = require(path.resolve(process.cwd(), 'src/utils/noting/is-note'));
+var isNote = require(path.resolve(process.cwd(), 'src/utils/noting/is-note-segment'));
```
Comment (robinedman) [inline test/unit/utils/noting/is-note-segment.spec.js]: Lacking `Segment`

**Comment by robinedman** [comment]
Good stuff! :+1:

**Comment by cutandpastey** [comment]
Ta!

**Pull request merged**